extension/src/entrypoints/popup/App.tsx

```diff
@@ -165,62 +179,278 @@ function App() {
       url: "/sounds/yippee.mp3",
     },
     {
-      label: "Music",
+      label: "Undertale Music",
       color: "cornflowerblue",
       emoji: "🎵",
       url: "/sounds/bg-music.mp3",
     },
   ];
 
+  function handleSync() {
+    console.log("This would sync!");
+  }
+
+  const [selectedFolder, setSelectedFolder] = useState("favorites");
+  const [folderSelectWidth, setFolderSelectWidth] = useState(0);
+
+  useEffect(() => {
+    // Resize the folder selector on value change
+    const canvas = document.createElement("canvas");
+    const context = canvas.getContext("2d");
+
+    if (context) {
+      context.font = `12px 'Instrument Sans', sans-serif`;
+      const textWidth = context.measureText(selectedFolder).width;
+      setFolderSelectWidth(textWidth + 48);
+    }
+  }, [selectedFolder]);
+
+  const [soundButtonOverflow, setSoundButtonOverflow] = useState("");
+
+  useEffect(() => {
+    // Since the height isn't calculated correctly when it loads, this forces it to recalculate
```

Comment:
I wonder if you might have some success with this in `useLayoutEffect`, maybe negating the need for the forceReflow timeout at least? https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen